### ISSUE TITLE
Cavern Crawl Bug

### ISSUE THREAD

**Issue by mvtabilitas** [ValveSoftware/Dota2-Gameplay#6311]
### Description

I don't know how this happened, but I hurt a hero that I did not have access to. I then proceeded into completing the wounded challenge and the bug persisted. I lost the rewards of the unfairly "unlocked" checkpoint but got the rewards of the fairly unlocked one, with a twist: I had to click the rewards on the map.
Everything in screenshots.

### Screenshots

![image](https://user-images.githubusercontent.com/74907147/210156886-1d65d8ec-fba8-43b3-8417-8727daa9d19e.png)
![image](https://user-images.githubusercontent.com/74907147/210156890-904bbd21-dfae-4da2-b419-0a45ee37fff9.png)
![image](https://user-images.githubusercontent.com/74907147/210156895-f373ebcf-f3c2-453a-a8a1-1c522627bf81.png)

**Comment by shaileshgarg** [comment]
Maybe the reward was not battle points but some Cavern Crawl tool, like Axe, Flare etc.

**Comment by mvtabilitas** [comment]
Indeed. It was weird anyway, after the game that completed the rogue challenge the cavern crawl screen appeared for a fraction of a second and then quickly disappeared. I am just reporting the problem without any requests.
